cfme/infrastructure/pxe.py

```diff
@@ -257,52 +369,115 @@ def get_pxe_image_type(self, image_name):
 
     @get_pxe_image_type.variant('ui')
     def get_pxe_image_type_ui(self, image_name):
-        navigate_to(self, 'All')
-        pxe_tree(self.name, 'PXE Images', image_name)
-        return pxe_details_page.pxe_image_type.text
+        view = navigate_to(self, 'Details')
+        view.sidebar.servers.tree.click_path('All PXE Servers', self.name,
+                                             'PXE Images', image_name)
+        details_view = self.create_view(PXESystemImageTypeDetailsView)
+        return details_view.entities.basic_information.get_text_of('Type')
 
     def set_pxe_image_type(self, image_name, image_type):
         """
         Function to set the image type of a PXE image
         """
+        # todo: maybe create appropriate navmazing destinations instead ?
         if self.get_pxe_image_type(image_name) != image_type:
-            navigate_to(self, 'All')
-            pxe_tree(self.name, 'PXE Images', image_name)
-            cfg_btn('Edit this PXE Image')
-            fill(pxe_image_type_form, {'image_type': image_type}, action=form_buttons.save)
+            view = navigate_to(self, 'Details')
+            view.sidebar.servers.tree.click_path('All PXE Servers', self.name,
+                                                 'PXE Images', image_name)
+            details_view = self.create_view(PXESystemImageTypeDetailsView)
```

Comment:
I guess you noticed my todo with similar statement. 
It will require me to create new object, add additional navigation steps for it, updated all related tests, etc and I am not sure it is worth doing this at least right now.
FA owner can do that later if it turns out that it makes sense.